test/backend/db_tests.js

```diff
@@ -450,6 +451,49 @@ module.exports = function(config, DB) {
         )
 
         test(
+          'locked accounts',
+          function (t) {
+            t.plan(5)
+
+            var lockedAt = Date.now()
+            var unlockCode = hex16()
+            var uid = Buffer(ACCOUNT.uid)
+            var email = Buffer(ACCOUNT.email)
+
+              // set lockedAt
+            return db.lockAccount(uid, { lockedAt: lockedAt, unlockCode: unlockCode })
+              .then(null, function(err) {
+                t.fail('We should not have failed this .lockAccount() request' + String(err))
+              })
+              .then(function(result) {
+                t.deepEqual(result, {}, 'Returned an empty object for lockAccount')
+                return db.account(uid)
+              })
+              .then(function(account) {
+                t.equal(account.lockedAt, lockedAt, 'account should now be locked')
+                return db.unlockCode(uid)
+              })
+              .then(function (_unlockCode) {
+                t.deepEqual(_unlockCode.unlockCode, unlockCode, 'unlockCode should be set')
+
+                // try to unlock the account
+                return db.unlockAccount(uid)
+              })
+              .then(null, function(err) {
+                t.fail('We should not have failed this .unlockAccount() request' + String(err));
```

Comment:
I never know what the right answer is here and I struggled with this myself in a previous issue related to deleting something. If you "unlock something that is already unlocked, then yeah, we're already done". ie. in my previous case, if I delete something which isn't there, we're done and we're succesful. Am just thinking out loud and not suggesting anything other than what you already have - just wondering what the best thing is here. An idempotent call would be nice here, but being explicit is also good. End of ramblings - feel free to ignore. :D